### DISCUSSION TITLE
Environment variables such as MIKRO_ORM_DB_HOST should be ignored when using clientUrl in the driver options

### DISCUSSION THREAD

**Discussion by sawa-ko** [mikro-orm/mikro-orm#4865] [Q&A]
Currently I want to do testing in my api and I have a special configuration for testing, but I also have an `.env` file in my project.

So in my special configuration for testing I have only one `clientUrl` assigned, but the database connection is overwritten by the environment variables in the .env file.

So, I would like to know if there is a way to do that when specifying a connection url to the driver and ignore the connection environment variables when using this option.

For example, if I use the `clientUrl` option, ignore the following environment variables:

```
MIKRO_ORM_DB_NAME
MIKRO_ORM_HOST
MIKRO_ORM_PORT
MIKRO_ORM_USER
MIKRO_ORM_PASSWORD
```

**Comment by B4nan** [comment]
It works in the opposite direction, env vars always have precedence. Also `clientUrl` is used as the base, so if you provide that and set a `user` or something else, it will take precedence over the value inferred from `clientUrl`.

I'd say you should have multiple `.env` files based on your environment and switch those instead of trying to do some overrides. Or if you want to override anything, override it via env vars instead - you can use `MIKRO_ORM_CLIENT_URL` there.
